src/structures/Message.js

```diff
@@ -714,15 +714,22 @@ class Message extends Base {
     return this.channel.send(data);
   }
 
+  /**
+   * Options for starting a thread on a message.
+   * @typedef {Object} StartThreadOptions
+   * @property {string} name The name of the new thread
+   * @property {ThreadAutoArchiveDuration} autoArchiveDuration The amount of time (in minutes) after which the thread
+   * should automatically archive in case of no recent activity
+   * @property {string} [reason] Reason for creating the thread
+   */
+
   /**
    * Create a new public thread from this message
    * @see ThreadManager#create
-   * @param {string} name The name of the new Thread
-   * @param {ThreadAutoArchiveDuration} autoArchiveDuration How long before the thread is automatically archived
-   * @param {string} [reason] Reason for creating the thread
+   * @param {StartThreadOptions} [options] Options for starting a thread on this message
    * @returns {Promise<ThreadChannel>}
    */
-  startThread(name, autoArchiveDuration, reason) {
+  startThread({ name, autoArchiveDuration, reason } = {}) {
```

Comment:
what about 
```suggestion
  startThread(options = {}) {
```
and then below
`create({ ...options, startMessage: this });`

Reply:
Thought of this but didn't know the best way to implement it, thanks